### ISSUE TITLE
SRG SSR bug : Failed to reload playlist

### ISSUE THREAD

**Issue by karlo2105** [streamlink/streamlink#475]
I didn't take enough time for testing yesterday, because I noticed now after about 30 seconds on all channels (SRF, RTS and RSI) and RTS sport streams, there is issue with playlist reloading.
Stream stops while looping endless.
Here is output for RTS Deux : `http://www.rts.ch/play/tv/direct#?tvLiveId=3608506`

`[stream.hls][warning] Failed to reload playlist: Unable to open URL: https://srg
ssruni10ch-lh.akamaihd.net/i/enc10uni_ch@191367/index_1200_av-p.m3u8?sd=6&dw=0&r
ebase=on&id=AgCy0TO8TW2D5%2fsbh1jy%2fZB2i43mTGrZJM2yXvCXp17DDbpjF7nJ1KhQ0FC0xqJp
hG0akaCfI0xs2A%3d%3d&hdnts=exp%3D1485249560~acl%3D%2Fi%2Fenc10uni_ch%40191367%2F
%2A~hmac%3D2095fa5cced5465adf2e5dd83a0ab1d2f213fad584dd93c0ea867a286aef0837 (403
 Client Error: Forbidden for url: https://srgssruni10ch-lh.akamaihd.net/i/enc10u
ni_ch@191367/index_1200_av-p.m3u8?sd=6&dw=0&rebase=on&id=AgCy0TO8TW2D5%2fsbh1jy%
2fZB2i43mTGrZJM2yXvCXp17DDbpjF7nJ1KhQ0FC0xqJphG0akaCfI0xs2A%3d%3d&hdnts=exp%3D14
85249560~acl%3D%2Fi%2Fenc10uni_ch%40191367%2F%2A~hmac%3D2095fa5cced5465adf2e5dd8
3a0ab1d2f213fad584dd93c0ea867a286aef0837&hdnts=exp%3D1485249560~acl%3D%2Fi%2Fenc
10uni_ch%40191367%2F%2A~hmac%3D2095fa5cced5465adf2e5dd83a0ab1d2f213fad584dd93c0e
a867a286aef0837)`

On the other side no problem with SRF and RSI sports events which are managed by swisstxt plugin. 
One additional address for RTS Sport 👍 

`"http://www.rts.ch/sport/direct/8222392-tennis.html" best
[Streamlink for Windows v0.2.0 - Git 5035b69]
[cli][info] Found matching plugin srgssr for URL http://www.rts.ch/sport/direct/
8222392-tennis.html
[plugin.srgssr][debug] Found rts video ID 8222392
error: Unable to open URL: http://il.srgssr.ch/integrationlayer/1.0/ue/rts/video
/play/8222392.json (502 Server Error: Bad Gateway for url: http://il.srgssr.ch/i
ntegrationlayer/1.0/ue/rts/video/play/8222392.json)
[End of Streamlink for Windows]`

Thanks

**Comment by beardypig** [comment]
Can you test out the latest version of the plugin on the [srgssr](https://github.com/beardypig/streamlink/tree/srgssr) branch @karlo2105? I was able to watch a stream for >1m. 

`http://www.rts.ch/sport/direct/8222392-tennis.html` should return no streams found instead of an error now :)

**Comment by karlo2105** [comment]
It's even worse now. 
`"http://www.rts.ch/play/tv/direct#?tvLiveId=3608506" best
[Streamlink for Windows v0.2.0 - Git 5035b69]
Failed to load plugin srgssr:
  File "imp.py", line 234, in load_module
  File "imp.py", line 172, in load_source
  File "<frozen importlib._bootstrap>", line 693, in _load
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "C:\Users\Ddr\Downloads\streamlink-portable-master\Releases\Streamlink\st
reamlink\plugins\srgssr.py", line 6, in <module>
    from streamlink.compat import urlparse, parse_qsl, urlencode, urlunparse
ImportError: cannot import name 'urlencode'`

**Comment by beardypig** [comment]
You'll need to update [src/streamlink/compat.py](https://raw.githubusercontent.com/beardypig/streamlink/srgssr/src/streamlink/compat.py) too. That change is already in the latest dev/nightly as well.

**Comment by karlo2105** [comment]
This morning I updated portable streamlink and I thought it should download the latest development version, that's why I didn't pay attention on compat.py. 
It is working now without errors. 
The link above was only active during match and printed same json errors as on first version that's why I posted it. Maybe it was just mislead. If I notice any non working redirections in future I would let you know. 
Thanks for fix. ;)
